class.jetpack-gutenberg.php

```diff
@@ -188,6 +188,20 @@ public static function set_extension_available( $slug ) {
 	 * @param array  $details A free-form array containing more information on why the extension is unavailable.
 	 */
 	public static function set_extension_unavailable( $slug, $reason, $details = array() ) {
+		/**
+		 * Filter 'jetpack_block_editor_enable_upgrade_nudge' with `true` to enable or `false`
+		 * to disable paid feature upgrade nudges in the block editor.
+		 *
+		 * @since 7.7.0
+		 *
+		 * @param boolean
+		 */
+		if ( 'missing_plan' === $reason && ! apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false ) ) {
+			// The block editor applies an upgrade nudge if `missing_plan` is the reason.
+			// Add a suffix to disable and provide informative reason.
+			$reason .= '__nudge_disabled';
```

Comment:
`'missing_plan' === $reason` is actually not a sufficient, but only a necessary criterion to be displayed: That's why blocks such as VideoPress currently don't display the nudge.

The other necessary criterion is the presence of a `required_plan` item in the `details` object the availability endpoint returns optionally for each unavailable block/plugin. We could thus also piggyback on that criterion.

Reply:
Updated comment in c2f0719d020e3a90894c7bdb030d56806a8006ef

Use _may apply_ instead of _applies_ to indicate that the reason isn't sufficient to ensure the nudge is shown, but changing the reason is significant to ensure that it isn't 🙂